src/java_bytecode/java_types.h

```diff
@@ -393,6 +397,46 @@ to_java_implicitly_generic_class_type(java_class_typet &type)
   return static_cast<java_implicitly_generic_class_typet &>(type);
 }
 
+/// \param type The type to check
+/// \return True if the given type is a generic class type or an implicitly
+/// generic class type, false otherwise
+inline const bool is_java_any_generic_class_type(const typet &type)
```

Comment:
I'm not sure this method improves readability - I wouldn't know what "any generic class" means so would have to look it up, probably would prefer to just use the condition. Or better yet if we can come up with a better name (I'll have a think)

Reply:
`is_class_with_generic_parameter`?

Reply:
One option would be to have `java_explicitly_generic_class_type` and `java_implicitly_generic_class_type` that would both extend `java_generic_class_type` (there is already a class like this but it's meaning now is only explicitly generic, not implicitly). Then we could use the word generic everywhere in the code with the expected meaning, only the lower level code would distinguish between the two.

Reply:
That sounds reasonable - is it a big change? If so, would it be reasonable to do a separate PR?

Reply:
I think it would make a good sized PR and the issue is independent of this ticket. I can create a Jira ticket for it.

Reply:
https://diffblue.atlassian.net/browse/TG-2639

Reply:
I think in the mean time (since we don't know when we're going to get to this TD) I'd rename this function (or eliminate it if you prefer) according to above suggestions